src/Dynode/model/core/socket/outputSocket.ts

```diff
@@ -37,75 +32,82 @@ class OutputSocket extends Socket {
    * @param parentNode  See {@link parent}
    * @param storageMode Option to enable StorageMode
    */
-  constructor(parentNode: Node, typeObject?: ValueType, storageMode?: boolean) {
-    super();
+  constructor(parentNode: Node, value?: T, ValueType?: ValueConstructor<T>, storageMode?: boolean) {
```

Comment:
Here you are using "ValueType" to denote the related value type class constructor. I support this naming convention, but you have to be consistent how you name these references to value type class constructors.

Reply:
valueType and ValueConstructor would be renamed SocketValue and SocketValueType accordingly.